sqlglot/dialects/exasol.py

```diff
@@ -19,6 +23,25 @@ def _sha2_sql(self: Exasol.Generator, expression: exp.SHA2) -> str:
     return self.func(func_name, expression.this)
 
 
+# https://docs.exasol.com/db/latest/sql_references/functions/alphabeticallistfunctions/trunc%5Bate%5D%20(datetime).htm
+# https://docs.exasol.com/db/latest/sql_references/functions/alphabeticallistfunctions/trunc%5Bate%5D%20(number).htm
+def _build_trunc(args: t.List[exp.Expression]) -> exp.Expression:
+    first, second = seq_get(args, 0), seq_get(args, 1)
+
+    if not first or not second:
+        return exp.Anonymous(this="TRUNC", expressions=args)
+
+    if not first.type:
+        first = annotate_types(first, dialect="exasol")
+
+    if first.is_type(exp.DataType.Type.DATE, exp.DataType.Type.TIMESTAMP) and isinstance(
+        second, exp.Literal
+    ):
```

Comment:
Sorry for the confusion, I meant that we could turn this into:


```suggestion
    if first.is_type(exp.DataType.Type.DATE, exp.DataType.Type.TIMESTAMP) and second.is_string
```

Because `is_string` already contains `isinstance(self, exp.Literal)`